packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -12,4 +12,32 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
    * Returns the Kaoto VSCode settings defined.
    */
   getVSCodeKaotoSettings(): Promise<ISettingsModel>;
+
+  /**
+   * Get metadata querying a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to retrieve the preferences
+   */
+  getMetadata<T>(path: string, key: string): Promise<T | undefined>;
+
+  /**
+   * Save preferences to a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to set the preferences
+   * @param metadata The preferences to be saved
+   */
+  setMetadata<T>(path: string, key: string, metadata: T): Promise<void>;
+
+  /**
+   * Retrieve resource content
+   * @param path The path of the resource, relative to the file being edited
+   */
+  getResourceContent(path: string): Promise<string | undefined>;
+
+  /**
+   * Save resource content
+   * @param path The path of the resource, relative to the file being edited
+   * @param content The content to be saved
+   */
+  saveResourceContent(path: string, content: string): Promise<void>;
```

Comment:
Should we declare the exception in case there is an exception while trying to save the file?

Reply:
It might be a good idea for the future, at this moment, I think whenever an exception happens, the notification and handling should be delegated to the host. For VSCode, this is what I think should happen:

1. A file is being saved
2. An exception is thrown from the `fs` or the MA module that controls the files being written
3. VSCode captures this exception and notifies the user through the VSCode alert functionality
4. We could choose to rethrow the exception to the Editor as well, just in case we want to either handle the UI part differently or notify the user as well.